pilot/pkg/xds/debug.go

```diff
@@ -669,6 +669,30 @@ func (s *DiscoveryServer) PushStatusHandler(w http.ResponseWriter, req *http.Req
 	_, _ = w.Write(out)
 }
 
+// PushContextDebug holds debug information for push context.
+type PushContextDebug struct {
+	AuthorizationPolicies *model.AuthorizationPolicies
+	NetworkGateways       map[string][]*model.Gateway
+}
+
+// PushContextHandler dumps the current PushContext
+func (s *DiscoveryServer) PushContextHandler(w http.ResponseWriter, req *http.Request) {
+	push := PushContextDebug{
+		AuthorizationPolicies: s.globalPushContext().AuthzPolicies,
```

Comment:
nit: can we get s.globalPushContext at the top of the function? To handle a rare case of push context updating while we call this